x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/notify_badge/notify_badge_with_api.tsx

```diff
@@ -24,6 +24,13 @@ export const RulesListNotifyBadgeWithApi: React.FunctionComponent<
   const [ruleSnoozeInfo, setRuleSnoozeInfo] =
     useState<RulesListNotifyBadgePropsWithApi['rule']>(rule);
 
+  // This helps to fix problems related to rule prop updates. As component handles the loading state via isLoading prop
+  // rule prop is obviously not ready atm so when it's ready ruleSnoozeInfo won't be updated without useEffect so
+  // incorrect state will be shown.
+  useEffect(() => {
+    setRuleSnoozeInfo(rule);
+  }, [rule]);
```

Comment:
This could be an indication of an anti-pattern:
https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Reply:
See this [comment](https://github.com/elastic/kibana/pull/153083#discussion_r1166456921).